services/storehost/storehost.go

```diff
@@ -1596,3 +1607,32 @@ func (t *StoreHost) ListExtents(tCtx thrift.Context) (res *store.ListExtentsResu
 
 	return res, nil
 }
+
+// UpgradeHandler implements the upgrade end point
+func (t *StoreHost) UpgradeHandler(w http.ResponseWriter, r *http.Request) {
+	t.logger.Info("Upgrade endpoint called on storehost")
+	// just report to controller as GOING_DOWN
+	t.SetNodeStatus(controller.NodeStatus_GOING_DOWN)
+	reporter := t.loadReporter.GetReporter()
+	// report as down
+	t.reportHostMetric(reporter, 1)
+
+	// wait for upgrade timeout
+	upgradeTimer := common.NewTimer(common.DefaultUpgradeTimeout)
+	defer upgradeTimer.Stop()
+	select {
+	case <-upgradeTimer.C:
+	}
```

Comment:
you don't need a select here, you could just do "<-upgradeTimer.C"; that said, instead of creating timer and stopping, etc, it would be better to do just call "time.Sleep".